src/vs/workbench/browser/layout.ts

```diff
@@ -229,6 +229,7 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 			wasPanelVisible: false,
 			transitionDisposables: new DisposableStore(),
 			setNotificationsFilter: false,
+			alwaysHideTitleBar: false,
```

Comment:
I would prefer if this was not a `boolean` but that it actualy held a value of the setting